src/streamlink/plugins/svtplay.py

```diff
@@ -87,18 +82,16 @@ def _get_live(self, path):
                 yield from DASHStream.parse_manifest(self.session, playlist['url']).items()
 
     def _get_vod(self):
-        res = self.session.http.get(self.url)
-        match = self.latest_episode_url_re.search(res.text)
-        if match:
-            res = self.session.http.get(
-                urljoin(self.url, match.group('url')),
-            )
-
-        match = self.vod_id_re.search(res.text)
-        if match is None:
+        vod_id = self._get_vod_id(self.url)
+
+        if vod_id is None:
+            res = self.session.http.get(self.url)
+            match = self.latest_episode_url_re.search(res.text)
+            vod_id = self._get_vod_id(match.group("url"))
```

Comment:
I added a guard. My body screamed at me to write a test because that is how I normally work professionally when someone points something out but I can't see that we are actually writing tests for the actual plugin logic, only for the plugin matcher. Is this correct or have I missed something?

Reply:
> I added a guard.

`vod_id = match and self._get_vod_id(match.group("url"))`
would've been fine too, as it doesn't require another code branch

> write a test

Most plugins don't have tests for stream data extraction, because it's almost impossible to keep up with website changes when plugins need to extract data from HTML documents or APIs. Then there's also geo restriction that comes into play, like in this case here. Those plugins which do have tests always use mocked data because of that, but it's rather pointless because mocking data responses doesn't represent the actual website content or APIs. If you want, you can add tests, but it's not necessary.

Reply:
I found that syntax you mentioned in some other plugins as well. I was looking at that syntax and though it was neat I didn't feel 100% comfortable with it so I went with what I felt like was the safest option because we don't write tests and it seems 50/50 if the plugin uses either syntax.

But now you just confirmed that it works the way I thought it would. So today I learned. I'll resolve the conversation now :)